test/unit/math/prim/fun/offset_multiplier_transform_test.cpp

```diff
@@ -65,9 +68,489 @@ TEST(prob_transform, offset_multiplier_f) {
                   stan::math::offset_multiplier_free(
                       stan::math::offset_multiplier_constrain(y, L, U), L, U));
 }
+
 TEST(prob_transform, offset_multiplier_f_exception) {
   double L = -10.0;
   double U = -27.0;
   EXPECT_THROW(stan::math::offset_multiplier_free(L - 0.01, L, U),
                std::domain_error);
 }
+
+TEST(prob_transform, offset_multiplier_constrain_matrix) {
+  Eigen::VectorXd x(4);
+  x << -1.0, 1.1, 3.0, 5.0;
+  Eigen::VectorXd sigma(4);
+  sigma << 1.1, 0.3, 6.0, 7.0;
+  Eigen::VectorXd offset(4);
+  offset << -2.0, 0.0, 0.2, 2.0;
+
+  double sigmad = 8.0;
+  double offsetd = -2.0;
+
+  Eigen::VectorXd sigma_bad(3);
+  Eigen::VectorXd offset_bad(3);
+
+  // matrix, real, real
+  {
+    Eigen::VectorXd result
+        = stan::math::offset_multiplier_constrain(x, offsetd, sigmad);
+    for (size_t i = 0; i < result.size(); ++i) {
+      EXPECT_FLOAT_EQ(result(i), stan::math::offset_multiplier_constrain(
+                                     x(i), offsetd, sigmad));
+    }
+    auto x_free = stan::math::offset_multiplier_free(result, offsetd, sigmad);
+    for (size_t i = 0; i < x.size(); ++i) {
+      EXPECT_FLOAT_EQ(x.coeff(i), x_free.coeff(i));
+    }
+  }
+  // matrix, matrix, real
+  {
+    Eigen::VectorXd result
+        = stan::math::offset_multiplier_constrain(x, offset, sigmad);
+    for (size_t i = 0; i < result.size(); ++i) {
+      EXPECT_FLOAT_EQ(result(i), stan::math::offset_multiplier_constrain(
+                                     x(i), offset(i), sigmad));
+    }
+    auto x_free = stan::math::offset_multiplier_free(result, offset, sigmad);
+    for (size_t i = 0; i < x.size(); ++i) {
+      EXPECT_FLOAT_EQ(x.coeff(i), x_free.coeff(i));
+    }
+    EXPECT_THROW(stan::math::offset_multiplier_constrain(x, offset_bad, sigmad),
+                 std::invalid_argument);
+  }
+  // matrix, real, matrix
+  {
+    Eigen::VectorXd result
+        = stan::math::offset_multiplier_constrain(x, offsetd, sigma);
+    for (size_t i = 0; i < result.size(); ++i) {
+      EXPECT_FLOAT_EQ(result(i), stan::math::offset_multiplier_constrain(
+                                     x(i), offsetd, sigma(i)));
+    }
+    auto x_free = stan::math::offset_multiplier_free(result, offsetd, sigma);
+    for (size_t i = 0; i < x.size(); ++i) {
+      EXPECT_FLOAT_EQ(x.coeff(i), x_free.coeff(i));
+    }
+    EXPECT_THROW(stan::math::offset_multiplier_constrain(x, offsetd, sigma_bad),
+                 std::invalid_argument);
+  }
+  // matrix, matrix, matrix
+  {
+    Eigen::VectorXd result
+        = stan::math::offset_multiplier_constrain(x, offset, sigma);
+    for (size_t i = 0; i < result.size(); ++i) {
+      EXPECT_FLOAT_EQ(result(i), stan::math::offset_multiplier_constrain(
+                                     x(i), offset(i), sigma(i)));
+    }
+    auto x_free = stan::math::offset_multiplier_free(result, offset, sigma);
```

Comment:
There's an `offset_multiplier_free` to go along with `offset_multiplier_constrain`.

These should be prim tests for `offset_multiplier_constrain`. Need some prim error tests for the free too once it's implemented but I was too lazy (the equivalent of [these](https://github.com/stan-dev/math/blob/develop/test/unit/math/prim/fun/lub_transform_test.cpp#L483)).

Reply:
I added the transform tests for matrices, lmk how those look!

Reply:
Cool beans, reviewing the indexing refactor for Stan right now but will get this in a bit.